File: backend/src/main/java/com/tribly/dto/teams/response/TeamListResponse.java
package com.tribly.dto.teams.response;

import com.tribly.dto.validation.ValidateSchema;
import java.util.List;
import org.eclipse.microprofile.openapi.annotations.media.Schema;

@Schema(description = "Paginated team list response")
@ValidateSchema
public record TeamListResponse(
    @Schema(description = "List of teams", required = true) List<TeamDetailDto> teams,
    @Schema(description = "Total number of teams", required = true) long total,
    @Schema(description = "Current page number", required = true) int page,
    @Schema(description = "Page size", required = true) int size) {}

File: frontend/src/api/dto/teamListResponse.ts
import type { TeamDetailDto } from './teamDetailDto.ts'

/**
 * Paginated team list response
 */
export interface TeamListResponse {
  /** List of teams */
  teams: TeamDetailDto[]
  /** Total number of teams */
  total: number
  /** Current page number */
  page: number
  /** Page size */
  size: number
}
